Icon, desktop, appdata and flatpakref fixes

krnowak opened the pull request:
The most important thing here is fixing the desktop file, so firefox shows up on the list of possible default web browser in gnome-control-center and installing an icon, so we do not show an ugly grey icon instead of the firefox logo in various places.

https://phabricator.endlessm.com/T16243

krnowak committed:
Drop .mozconfig

This file was likely used to configure the firefox project when
building it from source. We do not do it anymore, instead we download
the prebuilt firefox from mozilla, so this file is not necessary.

That file even had a typo in the last flag - extemsion instead of
extension…

krnowak committed:
Fix copy-pasta from vlc in the flatpakref file

Tsk tsk.

krnowak committed:
Add missing MimeType field in the desktop file

Apparently GAppInfo ignores desktop files without this field, so
firefox was not showing up in the default app chooser in gnome control
center.

krnowak committed:
Allow firefox to set itself as a default browser

It does it by talking to GConf over D-Bus, so we need to allow firefox
to talk to it.

That does not fully work yet, somehow firefox does not detect that it
is not a default browser anymore when I change this in
gnome-control-center. It probably tries to change wrong settings…

krnowak committed:
Fix icon name in the desktop file

It missed the update from the commit that changed org.mozilla.firefox
to org.mozilla.Firefox.

krnowak committed:
Install appdata file

It was never installed, which explains why some errors in the appdata
file went unnoticed before… The file will be fixed in the following
commit.

krnowak committed:
Fix appdata file

It will be fully fixed when the screenshot.png actually becomes a part
of firefox-flatpak repository.

krnowak committed:
Install an icon

Earlier we were taking an icon from the git repo - there is a
browser/branding directory containing the icons. But since we simply
download a prebuilt firefox, we don't get the icon anymore. Also, the
original icon does not fix the endless theme, so now we are installing
the firefox icon the eos-shell-content.

joaquimrocha commented on org.mozilla.Firefox.appdata.xml:
```diff
@@ -11,7 +12,7 @@
   <screenshots>
     <screenshot type="default">
-      <image>http://www.codeblocks.org/images/stories/imagebrowser/scr1.png</image>
+      <image>https://raw.githubusercontent.com/endlessm/firefox-flatpak/master/screenshot.png</image>
```
The URL should rather be: https://d3lapyynmdp1i9.cloudfront.net/screenshots/org.mozilla.Firefox/C/org.mozilla.firefox-screenshot1.jpg

joaquimrocha commented on org.mozilla.Firefox.json:
```diff
@@ -165,6 +165,11 @@
         },
         {
           "type": "file",
+          "url": "https://github.com/endlessm/eos-shell-content/raw/master/icons/bundle/64x64/apps/eos-app-org.mozilla.Firefox.png",
```
The URL should be added to the .in manifest as well. It's not being used right now but we should start using it.

joaquimrocha reviewed (changes requested):
Nice fixes @krnowak !
Just a couple of comments and also I think it's a good opportunity to fix this thing of having two json manifests. We should remove the final .json and use the .in. For that we should also replace the _branch_ value by _"@BRANCH@"_ as the Android Studio does.
Can you include these changes?

krnowak commented:
I have not removed the `.json` file in favor of just keeping `.json.in` only yet. From what I have seen, that `@BRANCH@` thing in android-studio flatpak is for the `runtime-version` field and yet you propose to put it for the `branch` field in firefox manifest. Isn't the `branch` field pointless in the manifest? So, I guess you wanted `"runtime-version": "@BRANCH@"` instead of `"branch": "@BRANCH@"`, which leads me to another question - the `@BRANCH@` substitution would be done by the `jenkins-build-flatpak-manifest` script which produces the file without the `.in` suffix, right? And yet, the `generate-json` script also takes an `.in` file to output the one without it. Do we need a `.in.in` file then there?

Also, regenerating the manifest file every time we fix some typo can be annoying, especially that it involves downloading all those `xpi` files to get their checksums, even when the firefox version did not change.

What I would propose is:
- Remove the `branch` field, the version is already stored in the generate-json script.
- Remove the `org.mozilla.Firefox.json.in` file.
- Modify the `generate-json` script so it changes the manifest file in place. It would simply replace all the `--extra-data` flags in the `finish-args` array with new ones. I'd rename the script then to something like `bump-version`.
- Replace the hard-coded value in the `runtime-version` field with the `@BRANCH@` substitution and rename the manifest file to have the `.in` suffix for the jenkins script to consume.

joaquimrocha commented:
Yeah, I meant the runtime-version.
The only problem with the approach you're suggesting is that if we want a placeholder in the _.in_ that is just for the //generate_json// (e.g. @MODULE_DEPENDENCIES@), then it could not be an in-place substitution. But we can address that problem when we have it. Let's do what you say please.

krnowak committed:
Remove the branch field

The branch field has the specific meaning - it describes the branch of
the application being flatpakked. It is also used for getting the base
app version if it is not explicitly specified.

We do not use any base app here nor is "52.0.1esr" a name of a branch
in mozilla's gecko-dev repository.

krnowak committed:
Make generate-json change the manifest in-place

This is so we can remove the .json.in file.

krnowak committed:
Remove the .json.in file

Nothing is using it at the moment - the generate-json script (soon to
be renamed) modifies the .json file in-place. Also, the two files here
unnecessarily adds the burden of keeping both of them in sync.

krnowak committed:
Rename the generate-json script

It's more of an updater tool now, than the generator.

krnowak committed:
Add a branch substitution for the jenkins-build-flatpak-manifest script

krnowak committed:
Update the JSON path in the updater script

krnowak committed:
Write a final newline when dumping json to file

So git will not complain that there is no newline at the end of file
after the update-firefox-version script has been run.

krnowak commented:
Updated.
